Sprint 1 + 2

tgillysuit opened the pull request:
-  Set up the backend server with Node.js and Express locally and on the VM.
-  Design and implement the database schema for MySQL.
- Create RESTful API endpoints for basic CRUD operations.
- Set up the React frontend using the Next.js framework with a basic layout and navigation.

tgillysuit committed:
Added some Goals to the README

tgillysuit committed:
Added a Next.js project and server folder

tgillysuit committed:
Added express and node_modules to the server folder. Also added a .gitattribute to help with potential diffs when adding and pushing to this repo.

tgillysuit committed:
Added a couple more packages to the Server folder. Started some code off for the server. Trying to go back through notes to understand building a server again along with the database. There is a .env.example file, the next dev will need to create their own .env file. Feel free to add more creds to the example file.

tgillysuit committed:
Added a frontend folder for the Next.js project to be placed in.

tgillysuit committed:
Added a sequelize package.

tgillysuit committed:
Created some code for the db.js file. An example schema.sql file for the database tables. A model for the schema to be created.'

RMarx1456 committed:
Added some structure to get us on the right track; some routers and controllers. Want to get the routers and controllers done next.

tgillysuit committed:
Added some things to the server.js file utilizing the dnd api that Tav gave us. Installed a few packages inside of the server folder, make sure to cd into that folder and do an npm i.

tgillysuit committed:
Did a few changes inside of ther server.js file on my main pc. Continuing to make some changes on my laptop.

RMarx1456 committed:
Cleaned up the code, and we have our approach now.

We want to setup a basic user/login system first, as further development (characters, roles) are entailed by that functionality.

Our next work will consist of building controllers and routers used for login functionality, and a front-end for easy accessibility. Also to note, I would rather have login success/failure returned in a constant time.

tgillysuit committed:
Linked Sequelize file, server file seems to work. Just have to link the server to the database.

tgillysuit committed:
Linked Sequelize file, server file seems to work. Just have to link the server to the database.

RMarx1456 committed:
Renamed controller/router for simplicity, modified server.js to reflect changes, added my own .env. DB is running on my end.

Controller, model, db, and routers should be worked on ASAP!

tgillysuit committed:
Added a schema.sql to be able to setup the database. I was running into some issues trying to setup the db on my main machine, but my laptop was working fine. I ended up figuring out what was going on and fixed the problem.

tgillysuit committed:
Updated the models.

tgillysuit committed:
Worked on some of the frontend logic inside of Next.js. Added a couple components where its a login form and registration form, as well as a parent component to handle the logic when a user selects register the registration form would pop up, but running into an error to why that state isn't working. Passing this off to Raymond for him to debug.

RMarx1456 committed:
Fixed registration-login form button display. Now we need to remove the old logic.

RMarx1456 committed:
Removed the old buttons. We should work on adding a bit of style.

RMarx1456 committed:
Fixed buttons, and made entrypoint the login form component. Styling and backend integration is needed now.

RMarx1456 committed:
Worked on some of the logic for submitting forms so default behavior is not assumed and the password does not end up in the query parameters. We need to get the backend interaction completed, and then style afterwards.

tgillysuit committed:
fixed a bug inside of the loginform component.

tgillysuit committed:
fixed a bug inside of the server.js file.

RMarx1456 committed:
Added a dependency autoinstall script.

Usage is "depdencies.sh" in CL, where CWD is DnD-Character-Sheet

tgillysuit committed:
Edited the shell script.

RMarx1456 committed:
Added npm dependencies to autoinstall in dependencies.sh

Will need to start getting server and FE up and running ASAP!

RMarx1456 committed:
Obviously I didn't check the name of the backend (actually /server). Fixed.

RMarx1456 committed:
Made the script echo some text output for which stage it is on.

RMarx1456 committed:
Let's see if this fixes node_modules in Dnd-Character-Sheet

RMarx1456 committed:
I'm trying to fix the frontend npm run dev on the vm. I changed Home() to Page() in app/page.js, and removed the unecessary package-lock.json in the wrong area.

RMarx1456 committed:
I'm trying to fix the entry points (layout.js, page.js); I consulted ChatGPT for the correct structure of them, so hopefully it will compile and not halt.

RMarx1456 committed:
Added a node auto-update to dependencies.sh. I'm upgrading node to v22.0.0 to hopefully fix the VM.

RMarx1456 committed:
Removing this package-lock.json AGAIN because it came back somehow.

RMarx1456 committed:
Got the VM FINALLY running the front-end at {VM_IP}:3000. I added some more bash scripts for DB instantiation and running the project in one command, so that can be worked on as I figure out how to set up the DB and henceforth, the backend itself.

RMarx1456 committed:
Progress on adding auto user addition to DB.

Am going to test this on another local linux WSL to see if it works.

RMarx1456 committed:
Added a injection warning to db_setup.sh

RMarx1456 committed:
Removed mysql secure setup from db_setup.sh

tgillysuit committed:
Created the UserController, UserRouter, and main router files.

tgillysuit committed:
Addeed the apiRouter to the server.js file. Pushing before testing to save the changes.

tgillysuit committed:
I've fixed a couple of bugs, implemented bcrypt for password hashing. Now to do some testing.

RMarx1456 committed:
Added echo to shwo what query is being executed.

tgillysuit committed:
testing a bug on the vm, updated the db_setup.sh file

tgillysuit committed:
fixing a bug in db_setup.sh

tgillysuit committed:
still fixing bugs in db_setup.sh

tgillysuit committed:
still fixing bugs in db_setup.sh

RMarx1456 committed:
Trying to fix line nine where the cd isn't found.

RMarx1456 committed:
Still trying to fix run_project.sh

tgillysuit committed:
fixed a bug inside of the RegistrationForm.jsx file

tgillysuit committed:
Merge branch 'main' of https://github.com/tgillysuit/DnD-Character-Sheet

tgillysuit committed:
fixed a bug inside of the RegistrationForm.jsx file

RMarx1456 committed:
Running server in run_project.sj

RMarx1456 committed:
Changed ordering of server-FE running; fixed filename error for cd (backend -> server).

RMarx1456 committed:
Added sleep between server running and FE running in run_project.sh

tgillysuit committed:
testing out a bug

tgillysuit committed:
Merge branch 'main' of https://github.com/tgillysuit/DnD-Character-Sheet

tgillysuit committed:
bugs bugs bugs bugs

tgillysuit committed:
oooo another bug

tgillysuit committed:
If I was a bug, I'd be terrified of myself.

tgillysuit committed:
testing something with the run_project.sh script and added in a dev: script inside of the package.json file. MAKE SURE TO npm i IF NOT SEEN.

tgillysuit committed:
did cd back into main folder to push changes.

tgillysuit committed:
I don't know what happened with express so reinstalled it into the server folder.

tgillysuit committed:
case sens is being weird with VM.

tgillysuit committed:
running into an error with package.json in the server folder on the VM. The package.json file is showing the dependencies inside of the scripts object instead of the dependencies object, but when pulled locally they're in the correct position.

tgillysuit committed:
reinstalled the dependencies inside of the server folder. Might have accidentally installed them outside of the frontend and backend folders.

RMarx1456 committed:
Added a DB creator in db_setup.sh.

RMarx1456 committed:
Trying to fix db_setup.sh.

RMarx1456 committed:
Renamed DB_DATABSE to DB_NAME in connect.js to hopefully correctly select the DB.

RMarx1456 committed:
Made the create DB into one line in db_setup.sh and tried fixing variable name error in RegistrationForm.jsx.

RMarx1456 committed:
Fixing alter perms in db_setup.sh.

RMarx1456 committed:
Fixing variable name in LoginForm.js (password: pw -> password: password)

RMarx1456 committed:
Added VM_IP import into RegistrationForm.jsx and LoginForm.jsx (imported variable is written to in vm_ip on usage of run_project.sh).

RMarx1456 committed:
Removing --turbopack from scripts to see if that reduces the ungodly compilation time.

RMarx1456 committed:
Trying to fix error 404 on fetching /login from backend VM.

RMarx1456 committed:
Changing back to sleep 5 in run_project.sh

RMarx1456 committed:
Added sanity test real quick for backend.

RMarx1456 committed:
Working on trying to fix CRUD responses. Then the CRUD setup will be bugfixed and completed for the MVP.

RMarx1456 committed:
Fixed router name in UserRouter.js

RMarx1456 committed:
Fixed router name in server.js

RMarx1456 committed:
Changed variable name in UserRouter.js to fix 400 response.

RMarx1456 committed:
Trying to fix sequelize scope error in UserController.js; removing .scope

RMarx1456 committed:
Fixing variable names again in UserController.js (credential -> username)

RMarx1456 committed:
Fixing RegistrationForm.jsx from pulling from the wrong port (FE port) and not BE port (Should have been through ${VM_IP}/signup}

RMarx1456 committed:
Adding bcrypt hashing to registration and fixing error 500 from UserController.js (now sets password_hash, as allow null is false)

RMarx1456 committed:
Added PM2 to the dependencies and script. Let's hope this works.

RMarx1456 committed:
Adding PM2 to PATH

RMarx1456 committed:
Fixing pm2 dependency.

RMarx1456 committed:
Trying to fix next.js dependency "next not found"

RMarx1456 committed:
Added npm i to dependences.sh for FE to see if that fixes next not found.

tgillysuit committed:
Merge branch 'main' of https://github.com/tgillysuit/DnD-Character-Sheet

RMarx1456 committed:
Removed VM_PASS from db_setup.sh

RMarx1456 committed:
Trying to fix next

RMarx1456 committed:
Trying to fix run_setup and pm2 and next

tgillysuit committed:
added some additional scripts to the db_setup.sh script

tgillysuit committed:
,erge branch 'main' of https://github.com/tgillysuit/DnD-Character-Sheet

tgillysuit committed:
fixing a minor syntax error inside of the db_setup cript.

tgillysuit committed:
fixed a bug inside of the connet.js file

tgillysuit committed:
changed a credential name inside of the db_setup.sh file

RMarx1456 committed:
Trying to fix dependencies again.

RMarx1456 committed:
Merge branch 'main' of https://github.com/tgillysuit/DnD-Character-Sheet

RMarx1456 committed:
Fixing dependencies.sh

RMarx1456 committed:
Still trying to fix dependencies.sh

RMarx1456 committed:
Fixing db connection (sync was writing bad queries).

RMarx1456 committed:
Trying to fix next AGAIN.

tgillysuit committed:
Fixed a few changes testing to see the database is linked backup again.

tgillysuit committed:
changed a little capitalization for the userSchema.js file.

tgillysuit committed:
changed a little capitalization for the shell script file.

tgillysuit committed:
testing something with the db shell script

tgillysuit committed:
created an admin folder with a couple blank files in there.

tgillysuit committed:
Created a couple branches, installed MUI, and added a NavBar component.

tgillysuit committed:
Added and debugged a few issues inside of the frontend.

RMarx1456 committed:
Huge commit. I need to commit because I haven't committed in a while. This is working on update and delete functionality.

RMarx1456 committed:
Replacing ID with username.

tgillysuit committed:
Added a User page with some functionality. Now to test it on the VM.

tgillysuit committed:
added functionality to the user

tgillysuit committed:
testing some different changes.

tgillysuit committed:
edited the readme.md

RMarx1456 committed:
Update README with next steps for project enhancement

Added next steps for project development and improvements.

RMarx1456 committed:
Merge pull request #1 from tgillysuit/user-up-del

Pull User Up Del branch into main

tgillysuit committed:
Updated the README.md

rifflere commented on README.md:
```diff
@@ -71,3 +71,79 @@ This application is designed for players of **Dungeons & Dragons 5th Edition (5E
+
+#### Group B (Tyler & Raymond)
+
+### Overall Goals
```
Helpful goal checklist!

AugleBoBaugles commented on server/server.js:
```diff
@@ -0,0 +1,74 @@
+    allowedHeaders: ['Content-Type', 'Authorization']
+}));
+
+// Middleware: log requests
```
Very helpful comments!

rifflere commented on server/server.js:
```diff
@@ -0,0 +1,74 @@
+    .catch(err => console.error('Error syncing database:', err));
+
+app.use('/api', apiRouter);
+app.use('/', loginRouter);
```
Is the default route a login? (No landing page)

rifflere commented on server/server.js:
```diff
@@ -0,0 +1,74 @@
+app.use('/', loginRouter);
+
+// Health check
+app.get("/api/health", (req, res) => {
```
I think this (and other get requests off of /api below) should be inside the apiRouter router

AugleBoBaugles replied:
This should be in a controller.

LauraV-702 commented on frontend/dnd-character-sheet/src/app/user/[username]/page.js:
```diff
@@ -0,0 +1,103 @@
+        return <Access />;
+    }
+
+    // TODO: Add some MUI Components down here to style it up.
```
Is MUI import or installed? :O

AugleBoBaugles commented on server/server.js:
```diff
@@ -0,0 +1,74 @@
+const characterInfo = new Set(['races', 'classes', 'backgrounds', 'monsters', 'spells', 'magicitems', 'weapons', 'armor']);
+
+// Open5e API proxy endpoints
+app.get("/api/dnd/:reference", async (req, res) => {
```
This also needs to be broken down into routers and controllers.

LauraV-702 commented on frontend/dnd-character-sheet/src/app/user/page.js:
```diff
@@ -0,0 +1,11 @@
+        <div>
+            <h2>User Access Denied</h2>
+            <p>You do not have permission to view this page.</p>
+        </div>
```
Good security! - Diana

LauraV-702 commented on frontend/dnd-character-sheet/src/components/LoginForm.jsx:
```diff
@@ -0,0 +1,84 @@
+                    email: data.user.email
+                });
+                
+                // Store in localStorage for persistence
```
For clarification, does this store and keep the user info permanently?

rifflere commented on server/server.js:
```diff
@@ -0,0 +1,74 @@
+import axios from 'axios';
+import sequelize from './database/connect.js'
+import apiRouter from './routers/Router.js';
+import loginRouter from './routers/UserRouter.js';
```
Nice use of middleware

AugleBoBaugles commented on server/controllers/UserController.js:
I like that this is broken down to just user actions. I assume there will also be EventController and LocationController. The naming conventions for the files are not clear to me though.

rifflere reviewed (commented):
Good use of MVC model!

LauraV-702 reviewed (commented):
Overall good stuff, but many files are separated and can be confusing to understand.

AugleBoBaugles reviewed (commented):
Documentation is well done!

RMarx1456 committed:
Working on some fixes in a new branch before I have to leave home. Using next/navigation for the navbar unless there's a better solution.

RMarx1456 committed:
Woooo!!!! Got a redirect working. (To /login)

RMarx1456 committed:
Got these working now!!! The navbar is complete now, asides from needing a homepage link.

RMarx1456 committed:
Fixed user deletion. Involved a very weird fix that required using "id.id" instead of just one id reference.

RMarx1456 committed:
Merge pull request #2 from tgillysuit/fix

Merge branch fix into main.

RMarx1456 committed:
Implemented update (Allowing the user to update their email).

A lot of the fixes made for this to happen were bandaid fixes and the localstorage object for the user itself is quote strange, as user details are encapsulated within the 1st level id field, as well as having the same info repeated in the first level as well.

RMarx1456 committed:
Merge pull request #3 from tgillysuit/fix

Implemented update (Allowing the user to update their email).
